rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2208,6 +2208,32 @@ void getProposedFederation_whenStorageProviderReturnsProposedFederation_shouldRe
         assertTrue(actualProposedFederation.isPresent());
         assertEquals(proposedFederation, actualProposedFederation.get());
     }
+        
+    @Test
+    void getProposedFederationSize_whenStorageProviderReturnsEmpty_shouldReturnNonExistentCode() {
+        // Arrange
+        int expectedCode = FederationChangeResponseCode.PROPOSED_FEDERATION_NON_EXISTENT.getCode();
+
+        // Act
+        int actualProposedFederationSize = federationSupport.getProposedFederationSize();
+
+        // Assert
+        assertEquals(expectedCode, actualProposedFederationSize);
+    }
+
+    @Test
+    void getProposedFederationSize_whenStorageProviderReturnsProposedFederation_shouldReturnProposedFederationSize() {
+        // Arrange
+        Federation proposedFederation = P2shErpFederationBuilder.builder().build();
+        storageProvider.setProposedFederation(proposedFederation);
+        int expectedSize = proposedFederation.getSize();
```

Comment:
```suggestion
        int expectedSize = 9;
```

same here, to make it deterministic

Reply:
Why is this not deterministic?